### PR TITLE
Logarithms

### PR DESCRIPTION
**Pull request by ChrisO345**
### Describe your change:

Added an algorithm to approximate b**? = x.

For example: 2**? = 32 --> 2**5 = 32 --> ? = 5

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: maths/logarithm.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    >>> logarithm(2, 0.9)
+    -6.578809999999997
+    >>> logarithm(9, 0.2)
+    -1.3652123999999999
+    >>> logarithm(0.9, 2)
+    -0.15200200000000003
+    """
+    
+    if base < 0 or end < 0:
+        raise ValueError('logarithm() takes 2 positive values')
```
Comment (cclauss) [inline maths/logarithm.py:32]: Please add tests where base is 0 and base is -1 so we can see how this function fails.

File: maths/logarithm.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            if _curr < end:
+                accuracy /= 10
+            else:
+                prev = start
+        else:
+            if _curr > end:
+                accuracy /= 10
+            else:
+                prev = start
+    return start
```
Comment (cclauss) [inline maths/logarithm.py:68]: In what circumstance would this line ever run?

File: maths/logarithm.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+
+if __name__ == "__main__":
+    print(logarithm(2, 9))
```
Comment (cclauss) [inline maths/logarithm.py:72]:
```suggestion
    print(f"{logarithm(2, 9) = }")
```
Repeat below.

**Review by ChrisO345 — COMMENTED** [review]
Added tests for base 0 and -1 and end 0 and -1

File: maths/logarithm.py (hunk 2 above)
Comment (ChrisO345) [reply maths/logarithm.py]: that line was originally left in for testing on my end. It has been removed now

**Comment by cclauss** [comment]
In preparation for that denial-of-service attack that is [Hacktoberfest](https://hacktoberfest.com), I am closing all pull requests that have failing tests.

**Pull request closed without merging**